subxt/src/book/usage/blocks.rs

```diff
@@ -27,13 +27,59 @@
 //! Aside from these links to other Subxt APIs, the main thing that we can do here is iterate over and
 //! decode the extrinsics in a block body.
 //!
-//! ## Example
+//! ## Decoding Extrinsics
 //!
-//! Given a block, you can [download the block body](crate::blocks::Block::extrinsics()) and iterate over
-//! the extrinsics stored within it. From there, you can decode the extrinsics and access various details,
-//! including the associated events:
+//! Given a block, you can [download the block body](crate::blocks::Block::extrinsics()) and [iterate over
+//! the extrinsics](crate::blocks::Extrinsics::iter()) stored within it. The extrinsics yielded, are of type
+//! [ExtrinsicDetails](crate::blocks::ExtrinsicDetails), which is just a blob of bytes that also stores which
+//! pallet and call in that pallet it belongs to. It also contains information about signed extensions that
+//! have been used for submitting this extrinsic.
+//!
+//! To use the extrinsic, you probably want to decode it into a concrete Rust type. These Rust types representing
+//! extrinsics from different pallets can be generated from metadata using the subxt macro or the CLI tool.
+//!
+//! When decoding the extrinsic into a static type you have two options:
+//!
+//! ### Statically decode the extrinsics into [the root extrinsic type](crate::blocks::ExtrinsicDetails::as_root_extrinsic())
```

Comment:
Perhaps it's also worth adding a small section on decoding extrinsics into dynamic types, to show that static decoding isn't the only option you have; right now that might not be obvious!